src/index.js

```diff
@@ -360,19 +359,14 @@ const compile = (schema, root, opts, scope, basePathRoot) => {
 
     /* Preparation and methods, post-$ref validation will begin at the end of the function */
 
-    const { type } = node
-    if (!type) enforceValidation('type is required')
-    if (type !== undefined && typeof type !== 'string' && !Array.isArray(type))
-      fail('Unexpected type')
-
-    const typeArray = type ? (Array.isArray(type) ? type : [type]) : ['any']
-    for (const t of typeArray) {
+    const typeArray =
+      node.type !== undefined ? (Array.isArray(node.type) ? node.type : [node.type]) : null
```

Comment:
Double negatives are confusing

```suggestion
      node.type === undefined ? null : (Array.isArray(node.type) ? node.type : [node.type])
```